src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -19,62 +20,46 @@
-        String indexName = findIndexWithDimensionAndType(pinecone, dimension, indexType);
-
-        return (indexName.isEmpty()) ? createNewIndex(pinecone, dimension, indexType) : indexName;
+        return new AbstractMap.SimpleEntry<>(indexName, pinecone);
     }
 
-    public static String findIndexWithDimensionAndType(Pinecone pinecone, int dimension, String indexType)
-            throws InterruptedException {
+    public static String findIndexWithDimensionAndType(Pinecone pinecone, int dimension, String indexType) {
         String indexName = "";
-        int i = 0;
         List<IndexModel> indexModels = pinecone.listIndexes().getIndexes();
         if(indexModels == null) {
             return indexName;
         }
-        while (i < indexModels.size()) {
-             IndexModel indexModel = isIndexReady(indexModels.get(i).getName(), pinecone);
+
+        for (IndexModel indexModel : indexModels) {
             if (indexModel.getDimension() == dimension
-                    && ((indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD)
-                        && indexModel.getSpec().getPod() != null
-                        && indexModel.getSpec().getPod().getReplicas() == 1
-                        && indexModel.getSpec().getPod().getPodType().equalsIgnoreCase("p1.x1"))
-                    || (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_SERVERLESS)))) {
+                && (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD) && indexModel.getSpec().getPod() != null)
+                || (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_SERVERLESS) && indexModel.getSpec().getServerless() != null)
+            ) {
                 return indexModel.getName();
             }
-            i++;
         }
         return indexName;
     }
 
-    public static String createNewIndex(Pinecone pinecone, int dimension, String indexType) {
+    public static String createNewIndex(Pinecone pinecone, int dimension, String indexType, boolean waitUntilIndexIsReady) throws InterruptedException {
         String indexName = RandomStringBuilder.build("index-name", 8);
         String environment = System.getenv("PINECONE_ENVIRONMENT");
         CreateIndexRequestSpec createIndexRequestSpec;
 
         if (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD)) {
-            CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
+            CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1" +
+                    ".x1");
             createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
         } else {
-            ServerlessSpec serverlessSpec = new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region(environment);
+            ServerlessSpec serverlessSpec =
+                    new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region("us-west-2");
```

Comment:
This needs to be either `us-west-2` for now, we shouldn't be relying on the `environment` value because it's often not correct in runs or when running manually.

Reply:
Can we leave a comment to explain the need to hard-code the environment

Reply:
Yes, good call. Public preview launched for serverless with one region `us-west-2` which is mostly hard-coded in tests at the moment.